Sources/Features/Shared/CloudKit/CloudKitError.swift

```diff
@@ -38,3 +52,148 @@ public struct CloudKitError: CloudKitErrorType {
         underlyingError = error
     }
 }
+
+
+
+// MARK: - Batch Modify Error Handling
+
+internal extension Array where Element: Equatable {
+
+    func bisect() -> (left: [Element], right: [Element]) {
+        let half = count / 2
+        return (Array(prefixUpTo(half)), Array(suffixFrom(half)))
+    }
+}
+
+internal extension BatchModifyOperationType where Save: Equatable, Save == Error.Save, Delete: Equatable, Delete == Error.Delete {
+
+    typealias ToModify = (toSave: [Save]?, toDelete: [Delete]?)
+    typealias ToModifyResponse = (left: ToModify, right: ToModify)
+
+    func bisect(error: Error) -> ToModifyResponse {
+        var left: ToModify = (.None, .None)
+        var right: ToModify = (.None, .None)
+
+        let remainingToSave = toSave?.filter { error.saved?.contains($0) ?? false }
```

Comment:
Condition should be something like: 
`let remainingToSave = toSave?.filter { !(error.saved?.contains($0) ?? false) }`

Also, just using the `contains` method on `error.saved` won't work properly. In the event that `saved` is an array of CKRecords (as in CKModifyRecordsOperation), what ought to be compared is the CKRecord.recordIDs. The saved CKRecords will _always_ be different from the initial CKRecords (as CloudKit returns the records-as-saved, which includes the updated system fields).

Thus, some ability to pass in a comparator function is needed?